src/components/LandingPageComponents/Stats.js

```diff
@@ -1,490 +1,465 @@
+    })
+  return asyncClient
 }
 
-
 const Stats = () => {
+  const [failedToUpdate, setFailedToUpdate] = useState(false)
+  const [tvl, setTvl] = useState(0)
+  const [swaprPrice, setSwaprPrice] = useState(0)
+  let [tx, setTx] = useState(0)
+  let [totalVolumeUSD, setTotalVolumeUSD] = useState(0)
+  const [isChartActive, setIsChartActive] = useState(false)
 
-    let [tvl, setTvl] = useState(0);
-    let [swaprPrice, setSwaprPrice] = useState(0);
-    let [tx, setTx] = useState(0);
-    let [totalVolumeUSD, setTotalVolumeUSD] = useState(0);
-    let [isChartActive, setIsChartActive] = useState(false);
-
-    useEffect(() => {
-        const tvlPromise = fetch('https://api.llama.fi/tvl/swapr')
-        tvlPromise.then((data) => {
-            return data.json();
-        }).then((decodedTvl) => {
-            setTvl((decodedTvl / 1000000).toFixed(1));
-        });
-    }, []);
+  useEffect(() => {
+    const tvlPromise = fetch('https://api.llama.fi/tvl/swapr')
+    tvlPromise
+      .then(data => {
+        return data.json()
+      })
+      .then(decodedTvl => {
+        setTvl((decodedTvl / 1000000).toFixed(1))
+      })
+  }, [])
 
-    useEffect(() => {
-        let coinCode = "arbitrum:0xde903e2712288a1da82942dddf2c20529565ac30"
-        const swaprPricePromise = fetch('https://coins.llama.fi/prices', {
-            method: 'POST',
-            body: JSON.stringify({
-                "coins": [
-                    coinCode
-                ]
-            })
-        });
-        swaprPricePromise.then((data) => {
-            return data.json();
-        }).then((decodedPrice) => {
-            let swaprPrice = decodedPrice.coins[coinCode].price.toFixed(3).toString();
-            setSwaprPrice(swaprPrice);
-        });
-    }, [])
+  useEffect(() => {
+    const coinCode = 'arbitrum:0xde903e2712288a1da82942dddf2c20529565ac30'
+    const swaprPricePromise = fetch('https://coins.llama.fi/prices', {
+      method: 'POST',
+      body: JSON.stringify({
+        coins: [coinCode]
+      })
+    })
+    swaprPricePromise
+      .then(data => {
+        return data.json()
+      })
+      .then(decodedPrice => {
+        const swaprPrice = decodedPrice.coins[coinCode].price.toFixed(3).toString()
+        setSwaprPrice(swaprPrice)
+      })
+  }, [])
 
-    useEffect(() => {
-        for (let val in apiUrlList) {
-            // eslint-disable-next-line
-            retrieveData(apiUrlList[val]).then((data) => {
-                let floatTx = parseFloat(data.data.swaprFactories[0].txCount);
-                let floatVolume = parseFloat(data.data.swaprFactories[0].totalVolumeUSD);
+  useEffect(() => {
+    for (const apiUrl in apiUrlList) {
+      // eslint-disable-next-line
+      retrieveData(apiUrlList[apiUrl]).then(data => {
+        if (data) {
+          let floatTx = parseFloat(data.data.swaprFactories[0].txCount)
+          let floatVolume = parseFloat(data.data.swaprFactories[0].totalVolumeUSD)
```

Comment:
Best to create to import the immediate subgraph clients from the source. Caching probably won't make sense here.

Reply:
Ok, what you are saying is to use `immediateSubgraphClients` instead of `subgraphClients` right?

So we use the `subgraphClients` to leverage caching? And the other in more simple cases?

I'll read more about Apollo client to understand this part of the system. If have any reading suggestions I appreciate that.

Reply:
Ok, I imported and replaced the local client, so we use `immediateSubgraphClients` directly instead of building an Apollo Client.